config_utilities/include/config_utilities/virtual_config.h

```diff
@@ -206,13 +206,14 @@ void declare_config(VirtualConfig<BaseT>& config) {
   auto data = internal::Visitor::visitVirtualConfig(config.isSet(), config.optional_, config.getType());
 
   // If setting values create the wrapped config using the string identifier.
+  const bool type_optional = config.optional_ || config.config_;
```

Comment:
Would it make sense to add a brief comment which case is handled here (for future reference)? I think this does the right thing but not immediately clear to me when reading that the config being set makes it optional.